source/community/release_notes/teak/dev_op_release_notes.rst

```diff
@@ -294,6 +294,81 @@ Removed Settings
+     remotely. This allows for additional security
+     restrictions, and the new code includes several security enhancements.
+   - *Purpose:* The darklaunch feature allows operators to gain confidence in preparing for a switch from local to remote 
+     codejail. When enabled, it can send all 
+     codejail executions to both local and remote codejail, while only using the results of the local execution and 
+     suppressing all errors from the remote side. 
+     This allows operators to discover issues in the remote service’s configuration under real production traffic
+     conditions.
+
+   - *Usage:* To use darklaunch to switch from local to remote:
+
+      - Create a codejail-service cluster
+      - Configure LMS and CMS to call it by configuring ``CODE_JAIL_REST_SERVICE_HOST`` but not ``ENABLE_CODEJAIL_REST_SERVICE`` (which must remain disabled for 
+        the moment).
+      - Begin the dark launch by setting ``ENABLE_CODEJAIL_DARKLAUNCH`` to true. Traffic will begin flowing to the new service, but the results will be ignored.
+         - The only user-visible impact should be that codejail executions take twice as long, as the local and remote executions are performed serially.
+      - Observe telemetry to discover errors and behavior mismatches.
+
+         - Mismatches can include:
+
+            - One side failed to execute entirely (“unexpected error”) while the other did not. This might include 
+              network issues.
+            - One side returned an error from the submitted code, while the other did not, or produced a different error.
+            - Both sides succeeded, but the returned globals dictionaries differed.
+
+         - Error and warning logs from ``safe_exec.py`` in edxapp containing ``codejail darklaunch`` will tell you about configuration problems, unexpected errors, and
+           mismatches in behavior between the two environments.
+         - Span-based telemetry (New Relic, Datadog, etc.) can be used to track rates of mismatches and break them down by course ID and type. See 
+           ``set_custom_attribute`` calls starting with ``codejail``. in 
+           `safe_exec.py <https://github.com/openedx/edx-platform/blob/release/teak.master/xmodule/capa/safe_exec/safe_exec.py>`_  
+           for available attributes. The local-only, remote-only and local/remote darklaunch calls all have different span names as well, e.g. 
+           ``safe_exec.remote_exec_darklaunch``.
+         - Use ``CODEJAIL_DARKLAUNCH_EMSG_NORMALIZERS`` to normalize away spurious mismatches between the environments. (Not all mismatches can be readily ignored, such 
+           as ordering differences in sets.)
+
+      - Once behavior and performance differences are resolved, remove ``ENABLE_CODEJAIL_DARKLAUNCH`` and set ``ENABLE_CODEJAIL_REST_SERVICE`` to true. This will complete 
+        the migration, and codejail executions will only be performed on the remote service.
+         
```

Comment:
@cmltaWt0 should we put an operator note about Credentials/Badging?

Reply:
@cmltaWt0 You can mention whatever you want :) I'm just the janitor putting it all together :) You can put the content you want as comments here inline, or add them to [the wiki](https://docs.openedx.org/en/latest/community/release_notes/teak/dev_op_release_notes.html).

Reply:
Ahh. Sure, let me add it to the wiki. Thanks!

Reply:
@cmltaWt0 this is the link to the wiki, please let us know when you've added an entry. https://openedx.atlassian.net/wiki/spaces/COMM/pages/4570710024/Teak+-+Operator+Dev+Notes

Reply:
@cmltaWt0 ping on this.